src/modules/user/user.usecase.ts

```diff
@@ -518,4 +555,36 @@ export class UserUseCases {
       password,
     });
   }
+
+  async updatePassword(
+    user: User,
+    updatePasswordDto: UpdatePasswordDto,
+  ): Promise<void> {
+    const { currentPassword, newPassword, newSalt, mnemonic, privateKey } =
+      updatePasswordDto;
+
+    if (user.password.toString() !== currentPassword) {
+      throw new UnauthorizedException();
+    }
+    await this.userRepository.updateById(user.id, {
+      password: newPassword,
+      hKey: Buffer.from(newSalt),
+      mnemonic,
+    });
+
+    const [keyServer] = await this.keyServerRepository.findUserKeysOrCreate(
+      user.id,
+      {
+        encryptVersion: privateKey,
```

Comment:
The encryptVersion should be sent by the front, which chooses how to encrypt the key (zero-knowledge), also, this field should not have the value of the privateKey, there is a field for that, as I have already said [here](https://github.com/internxt/drive-server-wip/pull/175#discussion_r1269689781). 

I also do not advise you to create anything, just find it. The creation of the keys is just enforced on one point: the login. Having it on multiple points could be a troublesome